lib/ConfirmationOverlay/index.js

```diff
@@ -13,10 +13,24 @@ const ConfirmationOverlay = ({
   confirm,
   cancel,
   confirmationText,
+  failureText,
   className,
   show
 }) => {
-  const [isSubmitting, handleSubmitWithLoader] = useLoader(confirm);
+  const [hasFailed, setHasFailed] = useState(false);
+  const setHasFailedTimeout = useRef(null);
+
+  const handleConfirm = async () => {
+    try {
+      await confirm();
+    } catch (error) {
+      setHasFailed(true);
+      setHasFailedTimeout.current = setTimeout(() => setHasFailed(false), 2000);
```

Comment:
🚩 Do we need to handle canceling the timeout when this component unmounts too?

I wonder if there's a hook that could be abstracted here, something like `useTimeout`, which you pass a function and a delay, and it handles the canceling logic for you. Maybe you could also pass it multiple functions to call.